rllib/evaluation/sampler.py

```diff
@@ -51,14 +55,33 @@ class _PerfStats:
     def __init__(self):
         self.iters = 0
         self.env_wait_time = 0.0
-        self.processing_time = 0.0
+        self.pre_processing_time = 0.0
+        self.storing_samples_in_buffer = 0.0
         self.inference_time = 0.0
+        self.action_processing_time = 0.0
+        self.get_ma_train_batch = 0.0
+        self.postprocess_trajectories_so_far = 0.0
 
     def get(self):
+        # Mean multiplicator (1000 = ms -> sec).
+        factor = 1000 / self.iters
         return {
-            "mean_env_wait_ms": self.env_wait_time * 1000 / self.iters,
-            "mean_processing_ms": self.processing_time * 1000 / self.iters,
-            "mean_inference_ms": self.inference_time * 1000 / self.iters
+            # Waiting for environment (during poll).
+            "total_env_wait_ms": self.env_wait_time,
+            "mean_env_wait_ms": self.env_wait_time * factor,
+            # Raw observation preprocessing.
+            "mean_pre_processing_ms": self.pre_processing_time * factor,
+            # Time spent storing/copying env/model data in the buffer.
+            "mean_storing_samples_in_buffer_ms":
+                self.storing_samples_in_buffer * factor,
+            # Computing actions through policy.
+            "mean_inference_ms": self.inference_time * factor,
+            # Processing actions (to be sent to env, e.g. clipping).
+            "mean_action_processing_ms": self.action_processing_time * factor,
+            # Legacy: "processing" = obs pre-processing + action processing.
+            "mean_processing_ms":
+                (self.action_processing_time + self.pre_processing_time) *
+                factor,
```

Comment:
This still seems too much. Can we cut it down to 4-5 stats at most please?

Reply:
done

Reply:
Need the "total_env_time" for the test case (to subtract the high-variance env time from the rest). I removed the old "mean_processing_ms", which was inaccurate anyways as it mixed the postprocessing of the observations with the preprocessing of the actions.